addons/website_mass_mailing/static/src/js/website_mass_mailing.js

```diff
@@ -71,84 +101,87 @@ publicWidget.registry.newsletter_popup = publicWidget.Widget.extend({
+                if (self.$target.attr('quick-open')) {
+                    self.massMailingPopup.open();
+                    self.$target.removeAttr('quick-open');
+                }
+            }));
         }
 
-        this._rpc({
-            route: '/website_mass_mailing/get_content',
-            params: {
-                newsletter_id: self.$target.data('list-id'),
-            },
-        }).then(function (data) {
-            if (data.content) {
-                $('<div></div>').append(data.content).appendTo(popupcontent);
-            }
-            self.$target.find('input.popup_subscribe_email').val(data.email || "");
-            self.redirect_url = data.redirect_url;
-            if (!self.editableMode && !data.is_subscriber) {
-                $(document).on('mouseleave', _.bind(self.show_banner, self));
-
-                self.$target.find('.popup_subscribe_btn').on('click', function (event) {
-                    event.preventDefault();
-                    self._onClickSubscribe();
-                });
-            } else { $(document).off('mouseleave'); }
-        });
+        return Promise.all(defs);
     },
-    _onClickSubscribe: function () {
-        var self = this;
-        var $email = self.$target.find(".popup_subscribe_email:visible");
 
-        if ($email.length && !$email.val().match(/.+@.+/)) {
-            this.$target.addClass('o_has_error').find('.form-control, .custom-select').addClass('is-invalid');
-            return false;
-        }
-        this.$target.removeClass('o_has_error').find('.form-control, .custom-select').removeClass('is-invalid');
-
-        this._rpc({
-            route: '/website_mass_mailing/subscribe',
-            params: {
-                'list_id': self.$target.data('list-id'),
-                'email': $email.length ? $email.val() : false,
-            },
-        }).then(function (subscribe) {
-            self.$target.find('#o_newsletter_popup').modal('hide');
-            $(document).off('mouseleave');
-            if (self.redirect_url) {
-                if (_.contains(self.redirect_url.split('/'), window.location.host) || self.redirect_url.indexOf('/') === 0) {
-                    window.location.href = self.redirect_url;
-                } else {
-                    window.open(self.redirect_url, '_blank');
-                }
-            }
-        });
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * @param {string} content
+     * @private
+     */
+     _dialogInit: function (content) {
+         var self = this;
+         var editableMode = this.editableMode;
+         this.massMailingPopup = new massMailingDialog(this, {
+             backdrop: !editableMode,
+             $content: $('<div/>').html(content),
+             $parentNode: editableMode ? this.$target : false,
```

Comment:
I am not sure, why is opened at different location depending on if it is in editable mode or not ?